src/module_2/Module_2_EDA_Ex1.md

```diff
@@ -0,0 +1,3152 @@
+      <td>16.0</td>
+    </tr>
+    <tr>
+      <th>3476</th>
+      <td>a725fbe5a75dce9540b3ed082270e4e357425949bae903...</td>
+      <td>Top Up</td>
+      <td>UKH</td>
+      <td>2022-01-12 13:38:49</td>
+      <td>2022-01-01</td>
+      <td>NaN</td>
+      <td>1.0</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>7.0</td>
+    </tr>
+    <tr>
+      <th>3505</th>
+      <td>257be7ae940425880bbb20bf162c2616b32881bf0a8bda...</td>
+      <td>Top Up</td>
+      <td>UKI</td>
+      <td>2021-06-10 15:00:55</td>
+      <td>2021-06-01</td>
+      <td>4.0</td>
+      <td>2.0</td>
+      <td>2.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>36.0</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+**f. What is the profile of the clients?**
+
+As previously mentioned, I will only be working with a subset of the clients; with those that have information regarding their household.
+
+
+```python
+family_codes = [col for col in users.columns if col.startswith('count')]
+familiy_inputs = users.count_people.dropna().count()
+
+household_info = {f"any_{col}" : (users[col]>0).sum() / familiy_inputs for col in family_codes}
+household_info
+```
+
+
+
+
+    {'any_count_people': 0.9938461538461538,
+     'any_count_adults': 15.323076923076924,
+     'any_count_children': 0.4,
+     'any_count_babies': 0.07076923076923076,
+     'any_count_pets': 0.40615384615384614,
+     'any_count_regulars': 4.455384615384616}
+
+
+
+
+```python
+# Households with either pets, children or babies
+(users[['count_children', 'count_babies', 'count_pets']].sum(axis=1) > 0).sum() / familiy_inputs
+```
+
+
+
+
+    0.6123076923076923
+
+
+
+We can observe that 40% of the houses have children, 7% have babies, and 40% have pets. Additionally, we can observe that 61% of the houses have either pets, children or babies. 
+
+
+```python
+print(familiy_inputs)
+print(len(users))
+print(familiy_inputs/len(users))
+```
+
+    325
+    4983
+    0.06522175396347582
+
+
+Nevertheless, it is worth mentioning that this subset only represents 6% of our clients, as we obly have family composition information for this small subset. 
+
+Now, we will see the NUTS1 distribution:
+
+
+```python
+plt.hist(users['user_nuts1'].dropna(), bins=30, edgecolor='black')
+plt.title('Histogram of NUTS1')
+plt.xlabel('NUTS1')
+plt.ylabel('Number of Users')
+plt.show()
+```
+
+
+    
+![png](Module_2_EDA_Ex1_files/Module_2_EDA_Ex1_123_0.png)
+    
+
+
+We can observe that most users come from UKI, and all of them are from UK.
+
+**g. Are our clients loyal, or do they only have one order?**
+
+
+```python
+orders['user_order_seq'].value_counts().plot(kind="bar")
+```
+
+
+
+
+    <Axes: xlabel='user_order_seq'>
+
+
+
+
+    
+![png](Module_2_EDA_Ex1_files/Module_2_EDA_Ex1_126_1.png)
+    
+
+
+Unfortunately, it appears that the majority of our users place only a single order and do not return. This pattern suggests potential areas for improvement in customer retention strategies.
```

Comment:
Given the inmmense increase in the last 2 months it would be nice to be a little bit more careful. Imagine that people buy durable goods (like cleaning) and the average frequency is 2 months between puchases. Most clients will not have had time.